### PR TITLE
Add sample: Front Door Premium (Preview) with Web Application Firewall and Microsoft-managed rule sets

### PR DESCRIPTION
**Pull request by johndowns**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add sample: Front Door Premium (Preview) with Web Application Firewall and Microsoft-managed rule sets

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

File: 201-front-door-premium-waf-managed/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/201-front-door-premium-waf-managed/CredScanResult.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-front-door-premium-waf-managed%2Fazuredeploy.json)  [![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-front-door-premium-waf-managed%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-front-door-premium-waf-managed%2Fazuredeploy.json)
```
Comment (bmoore-msft) [inline 201-front-door-premium-waf-managed/README.md:13]: remove extraneous button

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-front-door-premium-waf-managed/README.md (hunk 1 above)
Comment (johndowns) [reply 201-front-door-premium-waf-managed/README.md]: Fixed.

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by johndowns** [comment]
@bmoore-msft Any other changes required for this one?

**Pull request merged**